src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2177,34 +2177,69 @@ var projectUniqueNamesForBuildIntToUpdate
                             logger);
                     }
 
+                    // raise Nuget batch start event
+                    var batchId = Guid.NewGuid().ToString();
+                    string name;
+                    nuGetProject.TryGetMetadata(NuGetProjectMetadataKeys.Name, out name);
+                    packageProjectEventArgs = new PackageProjectEventArgs(batchId, name);
+                    BatchStart?.Invoke(this, packageProjectEventArgs);
+                    PackageProjectEventsProvider.Instance.NotifyBatchStart(packageProjectEventArgs);
+
                     try
                     {
                         if (msbuildProject != null)
                         {
                             //start batch processing for msbuild
                             await msbuildProject.ProjectSystem.BeginProcessingAsync();
-
-                            // raise Nuget batch start event
-                            var batchId = Guid.NewGuid().ToString();
-                            string name;
-                            nuGetProject.TryGetMetadata(NuGetProjectMetadataKeys.Name, out name);
-                            packageProjectEventArgs = new PackageProjectEventArgs(batchId, name);
-                            BatchStart?.Invoke(this, packageProjectEventArgs);
-                            PackageProjectEventsProvider.Instance.NotifyBatchStart(packageProjectEventArgs);
                         }
 
                         foreach (var nuGetProjectAction in actionsList)
                         {
-                            executedNuGetProjectActions.Push(nuGetProjectAction);
                             if (nuGetProjectAction.NuGetProjectActionType == NuGetProjectActionType.Uninstall)
                             {
+                                executedNuGetProjectActions.Push(nuGetProjectAction);
+
                                 await ExecuteUninstallAsync(nuGetProject,
                                     nuGetProjectAction.PackageIdentity,
                                     packageWithDirectoriesToBeDeleted,
                                     nuGetProjectContext, token);
                             }
-                            else
+
+                            if (nuGetProjectAction.NuGetProjectActionType == NuGetProjectActionType.Uninstall)
```

Comment:
This should be merged with the if block above, it looks like the same condition

Reply:
yeah, that was silly :)